### PR TITLE
feature/anchor processing

### PR DESCRIPTION
**Pull request by gratcliff**
## 🧰 What's being changed?

The changes in this PR are twofold:
1. Anchor generation is now using github's slug algo to determine and tack on anchor `id`s within our header component
2. We are also exporting this information, so our `processor.ast` export method now returns `hAST`, which includes this information in `properties`

## 🧪 Testing
This can be tested by sending markdown to two methods: `exports.react()` and `exports.ast()`.
## 🗳 Checklist

* [X ] **🆕 I'm adding something new!** - Feature consistency
* [X ] **📸 I've made some changes to the UI!** - Technically new `id`s in the UI.

### CODE REVIEW

File: packages/markdown/components/Heading/index.jsx [hunk 1]
```diff
@@ -1,44 +1,20 @@
-/* eslint-disable consistent-return */
 const React = require('react');
 const PropTypes = require('prop-types');
 
-function generateHeadingId(e, anchors) {
```
Comment (rafegoldberg) [inline packages/markdown/components/Heading/index.jsx:5]: Very happy to see this go!

File: packages/markdown/index.js [hunk 2]
```diff
@@ -207,13 +209,15 @@ export function html(text, opts = {}) {
...
  */
 export function ast(text, opts = {}) {
   if (!text) return null;
   [text, opts] = setup(text, opts);
 
-  return processor(opts).use(remarkStringify, opts.markdownOptions).parse(text);
+  const rdmd = processor(opts);
+  const node = rdmd.parse(text);
+  return rdmd.runSync(node);
 }
```
[the remark this answers is not in the document]
Comment (gratcliff) [reply packages/markdown/index.js:224]: I'm into it!

**Comment by erunion** [comment]
@rafegoldberg @gratcliff This good to go?

**Comment by rafegoldberg** [comment]
Gabe should probably speak more to this, but I’m happy with this work! Looks like it needs a rebase, but otherwise...

**Comment by gratcliff** [comment]
> Gabe should probably speak more to this, but I’m happy with this work! Looks like it needs a rebase, but otherwise...

Mmm good point, let me handle that quickly. @erunion good on my end otherwise.

**Comment by erunion** [comment]
Will get a release PR going before I bounce for PTO.

**Comment by gratcliff** [comment]
> Will get a release PR going before I bounce for PTO.

Thank you! We're all set. Take another once-over if ya got time please!

**Pull request merged**